CMakeLists.txt

```diff
@@ -767,17 +760,8 @@ if(BUSTED_LUA_PRG)
     set_target_properties(functionaltest-lua PROPERTIES FOLDER test)
 endif()
 
-#add uninstall target
-if(NOT TARGET uninstall)
-    configure_file(
-        "cmake/UninstallHelper.cmake.in"
-        "${CMAKE_CURRENT_BINARY_DIR}/UninstallHelper.cmake"
-        IMMEDIATE @ONLY)
-
-    add_custom_target(uninstall
-        COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/UninstallHelper.cmake)
-endif()
-
+add_custom_target(uninstall
+  COMMAND ${CMAKE_COMMAND} -P ${PROJECT_SOURCE_DIR}/cmake/UninstallHelper.cmake)
```

Comment:
@Borwe was there a particular reason you chose to do this as a `*.in` file ? https://github.com/neovim/neovim/pull/14149 does not explain.

Reply:
This is taken straight from the [cmake FAQ](https://gitlab.kitware.com/cmake/community/-/wikis/FAQ#can-i-do-make-uninstall-with-cmake). This particular code snippet has existed for a VERY long time (since around 2010). The `IMMEDIATE` for `configure_file` is an undocumented keyword kept for backwards compatibility. It was officially "removed" in 2.x. It's so long ago that finding information about it is difficult. I think it's one of those "Well, no need to change it if it works" situations taken to the extreme.

Reply:
perfect! would you mind including that in the commit message?

Reply:
Good idea.